File: laboratory-nacos-client-one/src/main/scala/com/sunshine/alibaba/cloud/handler/FlowRuleHandler.scala
package com.sunshine.alibaba.cloud.handler

import com.alibaba.csp.sentinel.slots.block.flow.FlowRule
import com.sunshine.alibaba.cloud.jooq.laboratory.Tables.FLOW_RULE
import org.springframework.beans.factory.annotation.Autowired

import scala.collection.JavaConverters._


class FlowRuleHandler @Autowired() (dslContext: DefaultDSLContext) {

  @Autowired
  private val dslContextOne: DefaultDSLContext = null

  def getRule(): List[FlowRule] = {
    val flowRuleList: java.util.List[FlowRule] = dslContext.select()
      .from(FLOW_RULE)
        .fetch()
        .into(classOf[FlowRule])
    flowRuleList.asScala.toList
  }
}

File: laboratory-nacos-client-one/src/main/java/com/sunshine/alibaba/cloud/service/FlowRuleHandler.java
package com.sunshine.alibaba.cloud.service;

import com.alibaba.csp.sentinel.slots.block.flow.FlowRule;
import org.jooq.DSLContext;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

import static com.sunshine.alibaba.cloud.jooq.laboratory.Tables.FLOW_RULE;

/**
 * @author <sunshine> mysunshinedreams@163.com
 * @since 2019-05-24
 */
@Service
public class FlowRuleHandler {

    @Autowired
    private DSLContext dslContext;

    public List<FlowRule> getRule() {
        return dslContext.select()
                .from(FLOW_RULE)
                .fetch()
                .into(FlowRule.class);
    }
}
